modules/kafka/src/main/java/com/devonfw/module/kafka/common/messaging/logging/impl/MessageListenerLoggingAspect.java

```diff
@@ -0,0 +1,156 @@
+package com.devonfw.module.kafka.common.messaging.logging.impl;
+
+import static brave.internal.HexCodec.toLowerHex;
+
+import java.lang.reflect.Method;
+import java.time.Instant;
+
+import javax.annotation.PostConstruct;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.record.TimestampType;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.annotation.AnnotationUtils;
+import org.springframework.core.annotation.Order;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.util.Assert;
+import org.springframework.util.ObjectUtils;
+
+import com.devonfw.module.kafka.common.messaging.trace.impl.MessageSpanExtractor;
+import com.devonfw.module.kafka.common.messaging.trace.impl.MessageTraceSupport;
+
+import brave.Span;
+import brave.Tracer;
+
+/**
+ * @author ravicm
+ *
+ */
+@Aspect
+@Order(0)
+public class MessageListenerLoggingAspect {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MessageListenerLoggingAspect.class);
+
+  @Autowired
+  private MessageLoggingSupport loggingSupport;
+
+  @Autowired
+  private ConsumerGroupResolver consumerGroupResolver;
+
+  /**
+   *
+   */
+  @Autowired(required = false)
+  protected Tracer tracer;
+
+  /**
+   *
+   */
+  @Autowired(required = false)
+  protected MessageSpanExtractor spanExtractor;
+
+  /**
+   * @param call
+   * @param kafkaRecord
+   * @return
+   * @throws Throwable
+   */
+  @Around("@annotation(org.springframework.kafka.annotation.KafkaListener) && args(kafkaRecord,..)")
+  public Object logMessageProcessing(ProceedingJoinPoint call, ConsumerRecord<String, String> kafkaRecord)
+      throws Throwable {
+
+    openSpan(kafkaRecord);
```

Comment:
Devon does not use b3 headers currently but a simple correlation ID (see com.devonfw.module.logging.common.api.LoggingConstants). Please align the implementation with that.

Reply:
for this am awaiting for the decision to remove the Tracer implementation or not .

Reply:
we have discussed in last architect meeting to keep tracer implementation.